Add previous and next buttons for doc navigation

lvoytek opened the pull request:
## Done
This change allows users to navigate through documentation pages by clicking a previous or next page button, making navigation faster when reading sequentially. The buttons show the name of the previous and upcoming pages, and the links correspond to the adjacent pages, regardless of section, with section pages included.

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/
    - Be sure to test on mobile, tablet and desktop screen sizes
- [List additional steps to QA the new features or prove the bug has been resolved]

## Issue / Card

Fixes #

## Screenshots
Here are some examples:
The end of the Server Documentation [munin page](https://ubuntu-com-12132.demos.haus/server/docs/tools-munin) will show:

![munin](https://user-images.githubusercontent.com/35419443/194167733-37f93ee5-89c1-4398-8e68-bdd278bb5c2d.png)

The [introduction](https://ubuntu-com-12132.demos.haus/server/docs) to the Server Documentation will show:

![intro](https://user-images.githubusercontent.com/35419443/194168010-9e18c604-6517-49d2-996c-2477bfa813c6.png)

Likewise, Ubuntu Core's [How-To home page](https://ubuntu-com-12132.demos.haus/core/docs/how-to) will show:

![corehowto](https://user-images.githubusercontent.com/35419443/194168275-44a199ac-2f77-4eee-b670-e579c1485a4e.png)

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

webteam-app commented:
Demo starting at https://ubuntu-com-12132.demos.haus

marinacastejon commented:
Hi @lvoytek , could you please provide us with more context?
- Why was this changed? Had a problem been identified? 
- Was any designer involved in the decision?
- Did you or the person who made this proposal look at the internal UX guidelines on pagination?
Any information you could provide would help us in the QA. 
Thanks!

@al-bakalova FYI

lvoytek commented:
Hi @marinacastejon,

I've been working on cleaning up the Server Guide for the past few months, focused mainly on order and format of the pages. One suggestion my team had was to add prev/next pagination make it easier to navigate through everything. This allows for a similar use case to the offline pdf server guide that we publish in which the user reads pages sequentially. 

The UX is based on the same style as the [Ubuntu blog](https://ubuntu.com/blog), but with page names instead of numbers. I didn't work with a designer or see the internal guidelines though. It's fine if the style needs to be updated or if this change ends up being discarded.

Thanks!

anthonydillon commented:
Personally, I think it works well and is kinda common. We do in fact has an exact pattern for this use: https://vanillaframework.io/docs/patterns/pagination#article-pagination

lvoytek commented:
@anthonydillon Thanks for the code reference, I can update the pull request to match that if preferred. Here are what the above pages look like in this style:

![munin2](https://user-images.githubusercontent.com/35419443/195185973-be7f2d45-ca13-47cb-8567-c7716f516f4b.png)


![intro2](https://user-images.githubusercontent.com/35419443/195186179-ba19b40f-727c-4ea3-992d-9ea55e81ca26.png)

![howto2](https://user-images.githubusercontent.com/35419443/195186506-254b065f-2ee3-4ce9-8c64-f550545727a3.png)

marinacastejon commented:
Hi @lvoytek , 
@al-bakalova and I took a look and chatted about this solution and its pros and cons. 
Firstly, it'd be better as Ant suggested if we used the existing Vanilla pattern for [article pagination](https://vanillaframework.io/docs/patterns/pagination#article-pagination). The main reasons: adding previous and next labels helps users understand that it is article pagination (vs a button with a random name that might be confused with an action-triggering button) and also, this pattern was created also considering the visual design and makes the content feel more balanced visually speaking. 

The second recommendation is about the logic. The pagination right now goes from one expanded section (of the side nav) to another. And if the user browses the page only by clicking on the Previous/Next buttons, they might not realise they are up to browsing a new section of the side nav. So maybe we can consider Previous/Next buttons only for the subpages within a selected section. The advantage of keeping the current logic is of course that it is faster if the user wants to go through all pages. I would take into account  how knowledgeable your typical users are and take a decision based on that. A third option could be that when the user is going to move to a new section, the next label could be updated to specify the new section and the page. 

I also found this case when I'm in the[ tutorials ](https://ubuntu-com-12132.demos.haus/server/docs/tutorials)so I'd recommend you also take a look at this particular case. I'm not sure why it's showing Introduction also in the previous-content button. 
![image.png](https://images.zenhubusercontent.com/61309e789df65d8124887eac/99f551a7-0eda-4ab4-8171-ba2796fce4d2)

![image.png](https://images.zenhubusercontent.com/61309e789df65d8124887eac/c126e259-de95-49b0-b8e9-6fc86652e1e6)

lvoytek commented:
Thanks for the help @marinacastejon. I updated the code with the new style and added additional info about the section for relevant pages. For example, the Tutorials page now shows:
![tutorials](https://user-images.githubusercontent.com/35419443/196766205-9dc576c0-1528-4baa-b02b-3d9efa3127ff.png)

marinacastejon commented:
Hi @lvoytek , I'm glad our comments helped. Could you please share another demo URL so that we QA it again? The one above doesn't work. Thanks

lvoytek commented:
@marinacastejon Sure, is there any way I can get the @webteam-app bot to rerun it?

marinacastejon commented:
Nice work @lvoytek , looks great, thank you

lyubomir-popov reviewed (changes requested):
Can we please wrap in a pstrip and put a hr above to give it a little more separation from thew content? 
![image](https://user-images.githubusercontent.com/2741678/197508106-c50462f9-24de-460d-9529-bb0079e16518.png)

lyubomir-popov commented on templates/templates/docs/shared/_docs.html:
```diff
@@ -100,6 +100,49 @@ <h3 class="p-side-navigation__heading">{{ nav_group.navlink_text }}</h3>
     <main class="col-9">
       {{ document.body_html | safe }}
 
+      <div class="p-article-pagination">
```
```suggestion
<div class="p-strip">
<hr class="u-no-margin--bottom">
      <div class="p-article-pagination">
```

lvoytek replied:
Suggestion looks good to me, thanks! Added with indentation

lyubomir-popov commented on templates/templates/docs/shared/_docs.html:
```diff
@@ -100,6 +100,49 @@ <h3 class="p-side-navigation__heading">{{ nav_group.navlink_text }}</h3>
+          {{- loop(page.children) }}
+        {%- endfor %}
+      </div>
+
```
```suggestion
</div>
```

lyubomir-popov replied:
^^ Added the above suggestion in the code

petesfrench reviewed (approved):
LGTM
